portfolio.js

```diff
@@ -388,12 +388,22 @@ function popupDesktop(btnid) {
   });
 }
 
-// form validation function
 const form = document.getElementById('formspree');
 const errorvalid = document.getElementById('errorvalid');
 const email2 = document.getElementById('email');
-let emailAddress;
+const { fullName } = form.elements;
+const textData = form.elements.messagetext;
+let emailAddress; let fullname; let
+  messagetext;
 function hasUppercaseLetter(str) { return str.toLowerCase() !== str; }
+function userDataSave() {
+  const userData = {
+    fullname,
+    email: emailAddress,
+    messagetext,
+  };
+  window.localStorage.setItem('userdata', JSON.stringify(userData));
+}
```

Comment:
- [ ] So far, you've done a good job 👏. However, the values entered by the user are not saved in local storage. Please correct this and ensure that all other project requirements are met before submitting it for a second review. 👌😊

### Expected behaviours:point_down: 
1. The contact form information is persistent to page reloads.
1. After filling the form, and not submitting it, the entered values should be saved in the browser.
1. When visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page.
1. If at the previous visit to the page, in the same browser, no input was entered in the form then the form inputs should remain empty.